src/NugetFeedDeploymentRepository/NugetPackageExtractor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.PackageExtraction;
+
+namespace Etg.Yams.NuGet.Storage
+{
+    public class NugetPackageExtractor
+    {
+        public async Task ExtractPackage(Stream fileStream, string outputDirectory)
+        {
+            var packagePathResolver = new FlatPackagePathResolver(outputDirectory);
+
+            await PackageExtractor.ExtractPackageAsync(
+                fileStream,
+                packagePathResolver,
+                new PackageExtractionContext(
+                    PackageSaveMode.Files,
+                    PackageExtractionBehavior.XmlDocFileSaveMode,
+                    new TraceLogger(),
+                    signedPackageVerifier: null),
+                CancellationToken.None);
+        }
+    }
+
+    /// <summary>
+    /// Extracts package directly to root directory.
+    /// </summary>
+    class FlatPackagePathResolver : PackagePathResolver
+    {
+        public FlatPackagePathResolver(string rootDirectory)
+            : base(rootDirectory, false)
```

Comment:
not sure what is false? A named parameter would help :)

Reply:
Done